### PR TITLE
Cleanup json related attributes

### PR DESCRIPTION
**Pull request by pouladpld**
This update adds contract resolver with snake case naming strategy so excessive code in json related attributes can be cleaned up.

### CODE REVIEW

File: src/Telegram.Bot/Types/InlineQueryResults/InlineQueryResultGif.cs [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
 using Newtonsoft.Json;
+using System.ComponentModel;
 
 namespace Telegram.Bot.Types.InlineQueryResults
 {
     /// <summary>
     /// Represents a link to an animated GIF file.
     /// By default, this animated GIF file will be sent by the user with optional caption.
     /// Alternatively, you can provide message_text to send it instead of the animation.
-    /// </summary>
```
Comment (pouladpld) [inline src/Telegram.Bot/Types/InlineQueryResults/InlineQueryResultGif.cs]: closing tag is missing

File: src/Telegram.Bot/Types/MaskPosition.cs [hunk 2]
```diff
@@ -12,21 +12,25 @@ public class MaskPosition
         /// <summary>
         /// The part of the face relative to which the mask should be placed. One of “forehead”, “eyes”, “mouth”, or “chin”.
         /// </summary>
+        [JsonProperty(Required = Required.Always)]
```
Comment (pouladpld) [inline src/Telegram.Bot/Types/MaskPosition.cs]: How come it was fine before by having opt in and without this attribute on properties?

File: src/Telegram.Bot/Types/ReplyMarkups/ReplyKeyboardMarkup.cs [hunk 3]
```diff
@@ -11,21 +11,19 @@ public class ReplyKeyboardMarkup : ReplyMarkup
...
-        [JsonProperty(PropertyName = "keyboard", Required = Required.Always)]
+        [JsonProperty(Required = Required.Always)]
         public KeyboardButton[][] Keyboard { get; set; }
 
         /// <summary>
         /// Optional. Requests clients to resize the keyboard vertically for optimal fit (e.g., make the keyboard smaller if there are just two rows of <see cref="KeyboardButton"/>). Defaults to <c>false</c>, in which case the custom keyboard is always of the same height as the app's standard keyboard.
         /// </summary>
-        [JsonProperty(PropertyName = "resize_keyboard", Required = Required.Default,
-            DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        [JsonProperty(Required = Required.Always, DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
```
Comment (pouladpld) [inline src/Telegram.Bot/Types/ReplyMarkups/ReplyKeyboardMarkup.cs]: Should be optional

File: src/Telegram.Bot/Types/ReplyMarkups/ReplyKeyboardMarkup.cs [hunk 4]
```diff
@@ -11,21 +11,19 @@ public class ReplyKeyboardMarkup : ReplyMarkup
...
         public bool ResizeKeyboard { get; set; }
 
         /// <summary>
         /// Optional. Requests clients to hide the keyboard as soon as it's been used. Defaults to <c>false</c>.
         /// </summary>
-        [JsonProperty(PropertyName = "one_time_keyboard", Required = Required.Default,
-            DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        [JsonProperty(Required = Required.Always, DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
```
Comment (pouladpld) [inline src/Telegram.Bot/Types/ReplyMarkups/ReplyKeyboardMarkup.cs]: Should be optional

File: src/Telegram.Bot/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,3 @@
+﻿using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Telegram.Bot.Tests.Integ")]
```
Comment (pouladpld) [inline src/Telegram.Bot/Properties/AssemblyInfo.cs]: This file could be removed. Our sys integ tests are meant to test the tg client the same way our users might work with it. That's not meant for unit testing.

File: src/Telegram.Bot/Types/MaskPosition.cs (hunk 2 above)
Comment (Tuscen) [reply src/Telegram.Bot/Types/MaskPosition.cs]: Maybe there aren't anyone using it. I don't know. But it's definitely wouldn't work without either property name, naming strategy or contract resolver. I've tested it, json.net leaves fields as is (in pascal case).

File: src/Telegram.Bot/Properties/AssemblyInfo.cs (hunk 5 above)
Comment (Tuscen) [reply src/Telegram.Bot/Properties/AssemblyInfo.cs]: oh, I left it accidentally, will remove it.

**Comment by pouladpld** [comment]
Thanks @Tuscen

**Pull request merged**